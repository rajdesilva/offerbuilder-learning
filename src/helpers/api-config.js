import { init as initApm } from "@elastic/apm-rum";

export const baseURL = {
  API_URL_BASE: process.env.REACT_APP_BASE_API_URL,
  LOGIN_URL: process.env.REACT_APP_LOGIN_URL,
  LOGOUT_URL: process.env.REACT_APP_LOGOUT_URL,
  PROFILE_URL: process.env.REACT_APP_USER_PROFILE_URL,
  HISTORY_URL: process.env.REACT_APP_USER_HISTORY_URL,
};

export const apm = initApm({
  // Set required service name (allowed characters: a-z, A-Z, 0-9, -, _, and space)
  serviceName: process.env.REACT_APP_APM_SERVICE_NAME,
  serverUrl: process.env.REACT_APP_APM_SERVICE_URL,

  // Set service version (required for sourcemap feature)
  serviceVersion: process.env.REACT_APP_APM_SERVICE_VERSION,
});