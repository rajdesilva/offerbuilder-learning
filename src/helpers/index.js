export * from './api-config';
export * from './history';
