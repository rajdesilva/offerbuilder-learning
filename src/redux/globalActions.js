export const globalActions = {
  'SET_SUCCESS': 'SET_SUCCESS',
  'SET_ERROR': 'SET_ERROR',
  'SET_LANGUAGE': 'SET_LANGUAGE',
  'CLEAR': 'CLEAR',
}