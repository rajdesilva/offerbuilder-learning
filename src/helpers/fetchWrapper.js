import { message } from "antd";
import { createIntl, createIntlCache } from "@formatjs/intl";
import * as languages from "../langs/";
import { baseURL } from "./api-config";
import { history } from "./history";

// This is optional but highly recommended
// since it prevents memory leak
const cache = createIntlCache();

// Create the `intl` object
const intl = createIntl(
  {
    // Locale of the application
    // TODO:
    locale: "en",
    // Locale of the fallback defaultMessage
    defaultLocale: "en",
    messages: languages["en"],
  },
  cache
);
const defaultRequest = {
  credentials: "include",
  headers: {
    "Content-Type": "application/json",
  },
};
export const fetchWrapper = (url, payload) => {
  payload = {
    ...defaultRequest,
    ...payload,
  };
  return fetch(url, payload).then(
    (response) => {
      // Check that the response is valid and reject an error
      // response to prevent subsequent attempt to parse json
      if (!response.ok) {
        if (response.status === 500) {
          message.error(
            intl.formatMessage({
              id: "nemo.error500msg",
            })
          );
        } else if (response.status === 503) {
          message.error(
            intl.formatMessage({
              id: "nemo.error503msg",
            })
          );
        } else if (response.status === 504) {
          message.error(
            intl.formatMessage({
              id: "nemo.error504msg",
            })
          );
        } else {
          message.error(
            intl.formatMessage({
              id: "nemo.somethingWentWrongPleaseTryAgain",
            })
          );
        }
        return;
      }
      if (response.status === 403) {
        history.push("/access-denied");
      }
      return response.json ? response.json() : response;
    },
    (error) => {
      /**
       * user url, payload and error here for sending to the logger.
       */
      if (error instanceof TypeError) {
        window.location.href = baseURL.API_URL_BASE + "redirect";
      }
      console.error("fetch error : ", error);
    }
  );
};
