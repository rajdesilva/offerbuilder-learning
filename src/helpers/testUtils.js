import React from "react";
import { createStore } from "redux";
import { Provider } from "react-redux";
import { render as rtlRender } from "@testing-library/react";
import { getValue } from "./utility";
import rootReducers from "../redux/rootReducers";
import * as languages from "../langs";
import { IntlProvider } from "react-intl";
const initialReducerState = {};
window.getValue = getValue;

window.google = {
  maps: {
    Marker: class {},
    Map: class {},
    LatLngBounds: class {},
    places: {
      Autocomplete: class {},
      AutocompleteService: class {},
      PlacesServiceStatus: {
        INVALID_REQUEST: "INVALID_REQUEST",
        NOT_FOUND: "NOT_FOUND",
        OK: "OK",
        OVER_QUERY_LIMIT: "OVER_QUERY_LIMIT",
        REQUEST_DENIED: "REQUEST_DENIED",
        UNKNOWN_ERROR: "UNKNOWN_ERROR",
        ZERO_RESULTS: "ZERO_RESULTS",
      },
      PlacesAutocomplete: {
        INVALID_REQUEST: "INVALID_REQUEST",
        NOT_FOUND: "NOT_FOUND",
        OK: "OK",
        OVER_QUERY_LIMIT: "OVER_QUERY_LIMIT",
        REQUEST_DENIED: "REQUEST_DENIED",
        UNKNOWN_ERROR: "UNKNOWN_ERROR",
        ZERO_RESULTS: "ZERO_RESULTS",
      },
    },

    MarkerClusterer: class {},
    Geocoder: class {},
  },
};
Object.defineProperty(window, "matchMedia", {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});

function render(
  ui,
  {
    initialState = initialReducerState,
    store = createStore(rootReducers, initialState),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    return (
      <Provider store={store}>
        <IntlProvider locale="en" messages={languages["en"]}>
          {children}
        </IntlProvider>
      </Provider>
    );
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

// re-export everything
export * from "@testing-library/react";

// override render method
export { render };
