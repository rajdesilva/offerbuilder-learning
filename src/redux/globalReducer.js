import { globalActions } from "./globalActions";
import produce from "immer";

const initialState = {
  error: "",
  success: "",
  language: "",
};
export const globalReducer = produce((draft, action) => {
  switch (action.type) {
    case globalActions.SET_ERROR:
      draft.success = "";
      draft.error = action.payload;
      return;
    case globalActions.SET_SUCCESS:
      draft.error = "";
      draft.success = action.payload;
      return;
    case globalActions.SET_LANGUAGE:
      draft.language = localStorage.getItem('language')
      return;
    case globalActions.CLEAR:
      return initialState;
    default:
      return draft;
  }
}, initialState);
