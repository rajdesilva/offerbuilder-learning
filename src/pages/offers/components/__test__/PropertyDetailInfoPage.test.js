import React from "react";
import { createStore } from "redux";
import { render, fireEvent } from "../../../../helpers/testUtils";
import { MemoryRouter } from "react-router-dom";
import { tempStore } from "./tempStore";
import PropertyDetailInfoPage from "../PropertyDetailInfoPage";

const localStorageMock = (() => {
  let store = {};

  return {
    getItem(key) {
      return store[key] || null;
    },
    setItem(key, value) {
      store[key] = value.toString();
    },
    removeItem(key) {
      delete store[key];
    },
    clear() {
      store = {};
    },
  };
})();

Object.defineProperty(window, "sessionStorage", {
  value: localStorageMock,
});

describe("PropertyDetailInfoPage component test", () => {
  beforeEach(() => {
    window.sessionStorage.clear();
    jest.restoreAllMocks();
  });
  test("PropertyDetailInfoPage should get property info from session storage and get a snapshot", () => {
    const store = createStore(() => ({ ...tempStore }));
    window.sessionStorage.setItem(
      "property-info",
      JSON.stringify({
        img: "https://res.cloudinary.com/seekda-dev/image/upload/w_1920,h_1080,c_fill,f_auto,fl_lossy,q_auto/development/DEMO_PKM_002/Screenshot_2020-03-06_at_5-17-37_PM.png",
        images: [
          "https://res.cloudinary.com/seekda-dev/image/upload/w_1920,h_1080,c_fill,f_auto,fl_lossy,q_auto/development/DEMO_PKM_002/Screenshot_2020-03-06_at_5-17-37_PM.png",
        ],
        propertyCode: "DEMO_PKM_002",
        lcn: true,
        supplier: "ntp",
        channel: "KOGNITIV_QA",
        hotelName: "Hotel Paradise PKM",
        remainingCapital: null,
        city: "Vienna",
        arrivalDate: "28.04.21",
        latitude: 48.19912,
        longitude: 16.349869,
        highestMargin: 0,
        lowestMargin: 0,
        b2bPrice: 142.86,
        b2cPrice: 142.86,
        margin: "",
        country: "Austria",
        rating: 5,
        description: "Featuring an open-air swimming pool, a fitness center",
        descriptions: [],
        trustyou: {
          id: "dc07b3c3-47d9-45a2-84be-369a864671d2",
          info: {
            name: "Rodeway Inn",
            reviews_count: 442,
            score: "66",
            score_description: "Poor",
            sources_count: 18,
            ty_id: "dc07b3c3-47d9-45a2-84be-369a864671d2",
          },
        },
        isParent: true,
        key: "DEMO_PKM_002-0true",
        alternativeProperties: null,
      })
    );
    const { container } = render(
      <MemoryRouter initialEntries={["/property/DEMO_PKM_002"]}>
        <PropertyDetailInfoPage />
      </MemoryRouter>,
      {
        store,
      }
    );
    expect(container).toMatchSnapshot();
  });

  test("PropertyDetailInfoPage - check if trust you, rating and review present for given property", () => {
    const store = createStore(() => ({ ...tempStore }));
    window.sessionStorage.setItem(
      "property-info",
      JSON.stringify({
        img: "https://res.cloudinary.com/seekda-dev/image/upload/w_1920,h_1080,c_fill,f_auto,fl_lossy,q_auto/development/DEMO_PKM_002/Screenshot_2020-03-06_at_5-17-37_PM.png",
        images: [
          "https://res.cloudinary.com/seekda-dev/image/upload/w_1920,h_1080,c_fill,f_auto,fl_lossy,q_auto/development/DEMO_PKM_002/Screenshot_2020-03-06_at_5-17-37_PM.png",
        ],
        propertyCode: "DEMO_PKM_002",
        lcn: true,
        supplier: "ntp",
        channel: "KOGNITIV_QA",
        hotelName: "Hotel Paradise PKM",
        remainingCapital: null,
        city: "Vienna",
        arrivalDate: "28.04.21",
        latitude: 48.19912,
        longitude: 16.349869,
        highestMargin: 0,
        lowestMargin: 0,
        b2bPrice: 142.86,
        b2cPrice: 142.86,
        margin: "",
        country: "Austria",
        rating: 5,
        description: "Featuring an open-air swimming pool, a fitness center",
        descriptions: [],
        trustyou: {
          id: "dc07b3c3-47d9-45a2-84be-369a864671d2",
          info: {
            name: "Rodeway Inn",
            reviews_count: 442,
            score: "66",
            score_description: "Poor",
            sources_count: 18,
            ty_id: "dc07b3c3-47d9-45a2-84be-369a864671d2",
          },
        },
        isParent: true,
        key: "DEMO_PKM_002-0true",
        alternativeProperties: null,
      })
    );
    const { getByTestId } = render(
      <MemoryRouter initialEntries={["/property/DEMO_PKM_002"]}>
        <PropertyDetailInfoPage />
      </MemoryRouter>,
      {
        store,
      }
    );
    expect(getByTestId("property-details-lcn")).toBeTruthy();
    expect(getByTestId("property-details-trust-you")).toBeTruthy();
    expect(getByTestId("property-trust-you-review")).toBeTruthy();
  });

  test("PropertyDetailInfoPage - check if images, trust you, ratings are not so empty image is shown and other fields are hidden for given property", () => {
    const store = createStore(() => ({ ...tempStore }));
    window.sessionStorage.setItem(
      "property-info",
      JSON.stringify({
        img: "",
        images: [],
        propertyCode: "DEMO_PKM_002",
        lcn: false,
        supplier: "ntp",
        channel: "KOGNITIV_QA",
        hotelName: "Hotel Paradise PKM",
        remainingCapital: null,
        city: "Vienna",
        arrivalDate: "28.04.21",
        latitude: 48.19912,
        longitude: 16.349869,
        highestMargin: 0,
        lowestMargin: 0,
        b2bPrice: 142.86,
        b2cPrice: 142.86,
        margin: "",
        country: "Austria",
        // rating: 5,
        description: "Featuring an open-air swimming pool, a fitness center",
        descriptions: [],
        trustyou: null,
        isParent: true,
        key: "DEMO_PKM_002-0true",
        alternativeProperties: null,
      })
    );
    const { getByTestId, queryByTestId } = render(
      <MemoryRouter initialEntries={["/property/DEMO_PKM_002"]}>
        <PropertyDetailInfoPage />
      </MemoryRouter>,
      {
        store,
      }
    );
    expect(
      getByTestId("DEMO_PKM_002-property-empty-image")
    ).toBeInTheDocument();
    expect(queryByTestId("property-details-lcn")).not.toBeInTheDocument();
    expect(queryByTestId("property-details-trust-you")).not.toBeInTheDocument();
    expect(queryByTestId("property-trust-you-review")).not.toBeInTheDocument();
  });

  test("PropertyDetailInfoPage - check if only one image is present other are hidden", () => {
    const store = createStore(() => ({ ...tempStore }));
    window.sessionStorage.setItem(
      "property-info",
      JSON.stringify({
        img: "https://res.cloudinary.com/seekda-dev/image/upload/w_1920,h_1080,c_fill,f_auto,fl_lossy,q_auto/development/DEMO_PKM_002/Screenshot_2020-03-06_at_5-17-37_PM.png",
        images: [
          "https://res.cloudinary.com/seekda-dev/image/upload/w_1920,h_1080,c_fill,f_auto,fl_lossy,q_auto/development/DEMO_PKM_002/Screenshot_2020-03-06_at_5-17-37_PM.png",
        ],
        propertyCode: "DEMO_PKM_002",
        lcn: true,
        supplier: "ntp",
        channel: "KOGNITIV_QA",
        hotelName: "Hotel Paradise PKM",
        remainingCapital: null,
        city: "Vienna",
        arrivalDate: "28.04.21",
        latitude: 48.19912,
        longitude: 16.349869,
        highestMargin: 0,
        lowestMargin: 0,
        b2bPrice: 142.86,
        b2cPrice: 142.86,
        margin: "",
        country: "Austria",
        rating: 5,
        description: "Featuring an open-air swimming pool, a fitness center",
        descriptions: [],
        trustyou: {
          id: "dc07b3c3-47d9-45a2-84be-369a864671d2",
          info: {
            name: "Rodeway Inn",
            reviews_count: 442,
            score: "66",
            score_description: "Poor",
            sources_count: 18,
            ty_id: "dc07b3c3-47d9-45a2-84be-369a864671d2",
          },
        },
        isParent: true,
        key: "DEMO_PKM_002-0true",
        alternativeProperties: null,
      })
    );
    const { queryByTestId } = render(
      <MemoryRouter initialEntries={["/property/DEMO_PKM_002"]}>
        <PropertyDetailInfoPage />
      </MemoryRouter>,
      {
        store,
      }
    );
    expect(queryByTestId("property-img-show-more-btn")).not.toBeInTheDocument();
    expect(queryByTestId("property-image-5")).not.toBeInTheDocument();
    expect(queryByTestId("property-image-4")).not.toBeInTheDocument();
    expect(queryByTestId("property-image-3")).not.toBeInTheDocument();
    expect(queryByTestId("property-image-2")).not.toBeInTheDocument();
    expect(queryByTestId("property-image-1")).not.toBeInTheDocument();
  });

  test("PropertyDetailInfoPage - check if back btn is clicked and page hidden", () => {
    const setItemSpy = jest.spyOn(window.sessionStorage, "setItem");
    const store = createStore(() => ({ ...tempStore }));
    window.sessionStorage.setItem(
      "property-info",
      JSON.stringify({
        img: "https://res.cloudinary.com/seekda-dev/image/upload/w_1920,h_1080,c_fill,f_auto,fl_lossy,q_auto/development/DEMO_PKM_002/Screenshot_2020-03-06_at_5-17-37_PM.png",
        images: [
          "https://res.cloudinary.com/seekda-dev/image/upload/w_1920,h_1080,c_fill,f_auto,fl_lossy,q_auto/development/DEMO_PKM_002/Screenshot_2020-03-06_at_5-17-37_PM.png",
        ],
        propertyCode: "DEMO_PKM_002",
        lcn: true,
        supplier: "ntp",
        channel: "KOGNITIV_QA",
        hotelName: "Hotel Paradise PKM",
        remainingCapital: null,
        city: "Vienna",
        arrivalDate: "28.04.21",
        latitude: 48.19912,
        longitude: 16.349869,
        highestMargin: 0,
        lowestMargin: 0,
        b2bPrice: 142.86,
        b2cPrice: 142.86,
        margin: "",
        country: "Austria",
        rating: 5,
        description: "Featuring an open-air swimming pool, a fitness center",
        descriptions: [],
        trustyou: {
          id: "dc07b3c3-47d9-45a2-84be-369a864671d2",
          info: {
            name: "Rodeway Inn",
            reviews_count: 442,
            score: "66",
            score_description: "Poor",
            sources_count: 18,
            ty_id: "dc07b3c3-47d9-45a2-84be-369a864671d2",
          },
        },
        isParent: true,
        key: "DEMO_PKM_002-0true",
        alternativeProperties: null,
      })
    );
    const { getByTestId } = render(
      <MemoryRouter initialEntries={["/property/DEMO_PKM_002"]}>
        <PropertyDetailInfoPage />
      </MemoryRouter>,
      {
        store,
      }
    );
    const backBtn = getByTestId("property-info-back-btn");
    fireEvent.click(backBtn);
    expect(setItemSpy).toHaveBeenCalledTimes(1);
  });
});
