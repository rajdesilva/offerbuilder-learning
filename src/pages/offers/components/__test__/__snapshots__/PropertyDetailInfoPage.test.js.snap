// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyDetailInfoPage component test PropertyDetailInfoPage should get property info from session storage and get a snapshot 1`] = `
<div>
  <div
    class="ant-row"
  >
    <div
      class="ant-col ant-col-24"
    >
      <button
        class="ant-btn ant-btn-text"
        data-testid="property-info-back-btn"
        type="button"
      >
        <span
          aria-label="arrow-left"
          class="anticon anticon-arrow-left"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="arrow-left"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M872 474H286.9l350.2-304c5.6-4.9 2.2-14-5.2-14h-88.5c-3.9 0-7.6 1.4-10.5 3.9L155 487.8a31.96 31.96 0 000 48.3L535.1 866c1.5 1.3 3.3 2 5.2 2h91.5c7.4 0 10.8-9.2 5.2-14L286.9 550H872c4.4 0 8-3.6 8-8v-60c0-4.4-3.6-8-8-8z"
            />
          </svg>
        </span>
      </button>
      <span>
        Hotel Paradise PKM
      </span>
    </div>
  </div>
  <div
    class="ant-row"
    style="margin: -4px -4px 4px -4px;"
  >
    <div
      class="ant-col ant-col-24"
      style="padding: 4px 4px 4px 4px;"
    >
      <div
        class="ant-image"
        data-testid="property-image-0"
      >
        <img
          class="ant-image-img"
          src="https://res.cloudinary.com/seekda-dev/image/upload/w_1920,h_1080,c_fill,f_auto,fl_lossy,q_auto/development/DEMO_PKM_002/Screenshot_2020-03-06_at_5-17-37_PM.png"
        />
      </div>
    </div>
  </div>
  <div
    class="ant-row"
  >
    <div
      class="ant-col ant-col-5"
    >
      <div
        data-testid="property-details-lcn"
      >
        Local Capital Network
      </div>
      <div>
        Address
      </div>
      <div
        data-testid="property-details-address"
      >
        Vienna
        ,Â 
        Austria
      </div>
      <div>
        Star rating
      </div>
      <ul
        class="ant-rate ant-rate-disabled"
        role="radiogroup"
        tabindex="-1"
      >
        <li
          class="ant-rate-star ant-rate-star-full"
        >
          <div
            aria-checked="true"
            aria-posinset="1"
            aria-setsize="5"
            role="radio"
            tabindex="-1"
          >
            <div
              class="ant-rate-star-first"
            >
              <span
                aria-label="star"
                class="anticon anticon-star"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="star"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 00.6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0046.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
                  />
                </svg>
              </span>
            </div>
            <div
              class="ant-rate-star-second"
            >
              <span
                aria-label="star"
                class="anticon anticon-star"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="star"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 00.6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0046.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
                  />
                </svg>
              </span>
            </div>
          </div>
        </li>
        <li
          class="ant-rate-star ant-rate-star-full"
        >
          <div
            aria-checked="true"
            aria-posinset="2"
            aria-setsize="5"
            role="radio"
            tabindex="-1"
          >
            <div
              class="ant-rate-star-first"
            >
              <span
                aria-label="star"
                class="anticon anticon-star"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="star"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 00.6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0046.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
                  />
                </svg>
              </span>
            </div>
            <div
              class="ant-rate-star-second"
            >
              <span
                aria-label="star"
                class="anticon anticon-star"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="star"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 00.6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0046.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
                  />
                </svg>
              </span>
            </div>
          </div>
        </li>
        <li
          class="ant-rate-star ant-rate-star-full"
        >
          <div
            aria-checked="true"
            aria-posinset="3"
            aria-setsize="5"
            role="radio"
            tabindex="-1"
          >
            <div
              class="ant-rate-star-first"
            >
              <span
                aria-label="star"
                class="anticon anticon-star"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="star"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 00.6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0046.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
                  />
                </svg>
              </span>
            </div>
            <div
              class="ant-rate-star-second"
            >
              <span
                aria-label="star"
                class="anticon anticon-star"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="star"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 00.6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0046.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
                  />
                </svg>
              </span>
            </div>
          </div>
        </li>
        <li
          class="ant-rate-star ant-rate-star-full"
        >
          <div
            aria-checked="true"
            aria-posinset="4"
            aria-setsize="5"
            role="radio"
            tabindex="-1"
          >
            <div
              class="ant-rate-star-first"
            >
              <span
                aria-label="star"
                class="anticon anticon-star"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="star"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 00.6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0046.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
                  />
                </svg>
              </span>
            </div>
            <div
              class="ant-rate-star-second"
            >
              <span
                aria-label="star"
                class="anticon anticon-star"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="star"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 00.6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0046.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
                  />
                </svg>
              </span>
            </div>
          </div>
        </li>
        <li
          class="ant-rate-star ant-rate-star-full"
        >
          <div
            aria-checked="true"
            aria-posinset="5"
            aria-setsize="5"
            role="radio"
            tabindex="-1"
          >
            <div
              class="ant-rate-star-first"
            >
              <span
                aria-label="star"
                class="anticon anticon-star"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="star"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 00.6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0046.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
                  />
                </svg>
              </span>
            </div>
            <div
              class="ant-rate-star-second"
            >
              <span
                aria-label="star"
                class="anticon anticon-star"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="star"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 00.6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0046.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
                  />
                </svg>
              </span>
            </div>
          </div>
        </li>
      </ul>
      <div>
        "
        Trust you
        "
         
        rating
      </div>
      <div
        class="ant-row"
      >
        <span
          data-testid="property-details-trust-you"
        >
          Poor
        </span>
        <span>
          66
        </span>
      </div>
      <div
        class="ant-row"
        data-testid="property-trust-you-review"
      >
        Based On
        Â 
        442
        Â 
        Reviews
      </div>
    </div>
    <div
      class="ant-col ant-col-1"
    />
    <div
      class="ant-col ant-col-18"
    >
      <div>
        Featuring an open-air swimming pool, a fitness center
      </div>
    </div>
  </div>
  <div
    class="ant-divider ant-divider-horizontal"
    role="separator"
  />
</div>
`;
